### ISSUE TITLE
Self hosting using docker and a slightly modified electrs implementation

### ISSUE THREAD

**Issue by henryperson** [Blockstream/esplora#339]
I'd like to self host the electrs API service for my own personal use. In particular, I'd like to modify electrs a little bit for my use case by changing some config values that aren't exposed to the command line.

I've seen [this stack overflow answer](https://bitcoin.stackexchange.com/questions/91746/configure-esplora-to-point-to-the-bitcoind-servers-ip-address/91859#91859), followed the [readme here](https://github.com/Blockstream/electrs) and [extra details in the original](https://github.com/romanz/electrs/blob/master/doc/usage.md) and it seems like the general recommendation for someone who wants to modify electrs is to make my changes and then build and run everything from the new version of electrs and not worry about esplora, which I don't need for my use case anyway. This is all working fine. However, I'd rather do it from docker because I've had trouble correctly daemonizing the electrs server, and I don't really want to reinvent the wheel on guaranteeing server uptime.

Is it possible to run exactly the docker process described [here](https://github.com/Blockstream/esplora#how-to-build-the-docker-image), given I assume it handles daemonizing and other relevant config, but just build electrs from a forked and slightly changed electrs repo?

Thanks!

**Comment by henryperson** [comment]
From doing a bit more research, it seems like this would be the process:
- Fork electrs and this repo
- Make the changes to electrs
- Modify [this line](https://github.com/Blockstream/esplora/blob/master/Dockerfile.deps#L49) and the line two below to clone the forked electrs repo and checkout the correct commit/branch
- Rebuild and push the esplora base per [this](https://github.com/Blockstream/esplora#build-new-esplora-base)
- Modify the first line of the [main esplora dockerfile](https://github.com/Blockstream/esplora/blob/master/Dockerfile) to point to the alternate base

Is that right?

**Comment by shesek** [comment]
Yes, the steps you outlined are correct, except that you don't have to push the esplora base image, its enough to have it available locally when building the main esplora docker image.
